assets/src/editor/events/dates-and-tickets-metabox/with-event-date-and-ticket-entity-filter-state.js

```diff
@@ -0,0 +1,113 @@
+/**
+ * External dependencies
+ */
+import PropTypes from 'prop-types';
+import { compose, createHigherOrderComponent } from '@wordpress/compose';
+import { withSelect } from '@wordpress/data';
+
+/**
+ * Internal imports
+ */
+import '../../editor.css';
+import {
+	withDatesListFilterState,
+	getFilteredDateEntitiesList,
+} from '../dates-and-times/editor-date/filter-bar';
+import {
+	withTicketEntitiesListFilterState,
+	getFilteredTicketEntitiesList,
+} from '../tickets/editor-ticket/filter-bar';
+import { withEventEntity, withEventDateEntities } from '../events/data';
```

Comment:
The hoc is named `withEventDateEntities` because it communicates that it returns date entities belonging to an event. `withDateEntities` simply indicates the hoc is returning date entities and I would probably do that for a hoc that retrieves a query string to indicate what date entities to return.

> but it appears you have dropped that in most places in this PR and simply gone with *Date* instead of *EventDate*.

When we return `eventDateEntities` or `eventDateEntitiesLoaded` from a hoc like this, it means that anything the hoc wraps _must_ have props named `eventDateEntities` or `eventDateEntitiesLoaded` so it reduces the reusability of this hoc on components that don't care _where_ dateEntities come from.   The principle here is the hoc name indicates what it returns and what qualifier it has for what it returns (i.e. it returns `dateEntities`, and the qualifier is it is `dateEntities` belonging to an event), and then the HOC could be used with any component that receives `dateEntities` and `dateEntitiesLoaded` as props.  That same component could be used with something like a `withTicketDateEntities` that also returns `dateEntities` and `dateEntitiesLoaded` with zero adjustment.  Here's an example:

I have a component that just outputs a list of Dates:

```jsx
export default const FeedMeDateEntities = ( { dateEntities, dateEntitiesLoaded } ) => {
	if ( ! dateEntitiesLoaded ) {
		return <div>Still getting dates yo!</div>;
	}

	return <ul>
		{
			dateEntities.map( ( dateEntity ) => {
				return <li key={ dateEntity.id }>{ dateEntity.name }</li>
			} )
		}
	</ul>
};
```

Now I can use this component with various date entity provider hocs. 
Here's an example using `withEventDateEntities`

```jsx
import { withEventDateEntities, FeedMeDateEntities } from '@eventespresso/components';

const EventDateList = withEventDateEntities( FeedMeDateEntities );

const MyCustomComponent = ( { eventEntity } ) => {
	return <>
		<h2>Dates for the Event { eventEntity.name }</h2>
		<EventDateList eventEntity={ eventEntity } />
	</>
};
```

Here's an example using `withTicketDateEntities`

```jsx
import { withTicketDateEntities, FeedMeDateEntities } from '@eventespresso/components';

const TicketDateList = withTicketDateEntities( FeedMeDateEntities );

const MyCustomComponent = ( { ticketEntity } ) => {
	return <>
		<h2>Dates for the Ticket { ticketEntity.name }</h2>
		<TicketDateList ticketEntity={ ticketEntity } />
	</>;
};
```

I think that illustrates the reusability when entity hocs return a consistent pattern.  Another reason for implementing a consistent pattern is that it potentially opens the door for automatically generating hocs and components for our models reducing the amount of code in our repo (of course it doesn't matter WHAT the naming pattern is for that, only that there is consistency).